example/CMakeLists.txt

```diff
@@ -60,6 +60,9 @@ test_sc_example(logging logging/logging.c)
 test_sc_example(test_shmem testing/sc_test_shmem.c)
 
 if(SC_HAVE_GETOPT_H)
```

Comment:
I'm wondering whether we need to check for this #define.
If it is not defined, we are building a replacement -- options should always work the same.

Reply:
In principle you are right, but currently in that regard the autotools and cmake build system are not equivalent:

- in the autotools build, there is some logic to define SC_PROVIDE_GETOPT symbol
- this logic does not exist in the cmake build (SC_PROVIDE_GETOPT is not controlled in cmake); so currently the cmake build only check for header getopt.h and use that info to decide if or if sc_options.c, etc are included in the build

maybe we could add the functionnality in the cmake build (in another MR).

Reply:
> Currently in cmake build, SC_PROVIDE_GETOPT is not defined, so only getopt.h can be used, that way sc_options.c is only included when SC_HAVE_GETOPT_H is ON.
> 
> maybe we could align autotoools and cmake build in another MR.

I agree, this will be  useful to change and make both systems `#define` the same.